.github/workflows/CI.yml

```diff
@@ -823,31 +838,35 @@ jobs:
             "$HOME/Library/Application Support/com.apple.TCC/TCC.db"
           )
 
-          sqlQuery="INSERT OR IGNORE INTO access VALUES($values);"
-
-          for dbPath in "${dbPaths[@]}"; do
-            echo "Column names for $dbPath"
-            echo "-------------------"
-            sudo sqlite3 "$dbPath" "PRAGMA table_info(access);"
-            echo "Current permissions for $dbPath"
-            echo "-------------------"
-            sudo sqlite3 "$dbPath" "SELECT * FROM access WHERE service='kTCCServiceScreenCapture';"
-            sudo sqlite3 "$dbPath" "$sqlQuery"
-            echo "Updated permissions for $dbPath"
-            echo "-------------------"
-            sudo sqlite3 "$dbPath" "SELECT * FROM access WHERE service='kTCCServiceScreenCapture';"
+          for value in "${values[@]}"; do
+            for dbPath in "${dbPaths[@]}"; do
+              echo "Column names for $dbPath"
+              echo "-------------------"
+              sudo sqlite3 "$dbPath" "PRAGMA table_info(access);"
+              echo "Current permissions for $dbPath"
+              echo "-------------------"
+              sudo sqlite3 "$dbPath" "SELECT * FROM access WHERE service='kTCCServiceScreenCapture';"
+              execute_sql_query "$value" "$dbPath"
+              echo "Updated permissions for $dbPath"
+              echo "-------------------"
+              sudo sqlite3 "$dbPath" "SELECT * FROM access WHERE service='kTCCServiceScreenCapture';"
+            done
           done
 
-      - name: Run tests
+      - name: Run tests directly
         id: test
         timeout-minutes: 10
+        working-directory:
+          /opt/local/var/macports/build/_Users_runner_work_Sunshine_Sunshine_ports_multimedia_Sunshine/Sunshine/work/build/tests
         run: |
-          sudo port test "Sunshine"
+          sudo port install doxygen
+          sudo port install graphviz
+          sudo ./test_sunshine --gtest_color=yes
```

Comment:
Is this to get around the threading issue? If so, could we miss that issue when the user is running Sunshine from macports?

Is sudo required to run the test?

Reply:
@ReenigneArcher so basically, I don't even know if it can be said is a threading issue, and not a macports issue. This `test_sunshine` runs perfectly in any other scenario, via brew, via IDE, via terminal, anything, but when executed from anywhere inside port, it thread locks. I've tried everything I had in mind running port test locally, couldn't fix it.

I tried removing it from the tests section in ports, and running it in `post-build`, it does the same, locks on any input API, probably because of `CFRunLoop` getting locked, but only in ports.

So instead, I'd rather run it manually. And it's running faster too. Given that the tests are built using ports anyway, seems fine to me.

Regarding the user facing the issue later, I created a filter in the ports file, to avoid tests with HID in the name, we may also do it for other scenarios we want to avoid testing on the user device end. Actually, do we need to force run tests on installation on user devices? I was thinking, this test should not run on user device anyway, it moves the user mouse, and if the user is using the mouse during the tests, the tests will randomly break anyway. Other tests may have similar issues, I'd rather just run tests on the CI, and allow users to run tests manually later.

Yep, sudo required, I want to see if I can remove this restriction, but it seems the tests that stuff write to files break when running without sudo.

Reply:
> Actually, do we need to force run tests on installation on user devices? I was thinking, this test should not run on user device anyway, it moves the user mouse, and if the user is using the mouse during the tests, the tests will randomly break anyway. Other tests may have similar issues, I'd rather just run tests on the CI, and allow users to run tests manually later.

We don't need to force it, and I don't think using the other method does that. It would be good if users could run the tests on their specific machine though, once we get more tests it might be helpful to use as a debugging tool. There would probably need to be a message to not move the mouse during the test or other notifications depending on what tests we end up adding... but user running tests would be completely optional. CI is the main focus.